api/all/src/main/java/io/opentelemetry/api/internal/ConfigUtil.java

```diff
@@ -33,10 +37,21 @@ private ConfigUtil() {}
    */
   public static String getString(String key, String defaultValue) {
     String normalizedKey = normalizePropertyKey(key);
+    Set<Map.Entry<String, String>> properties =
+        new HashSet<>(System.getProperties().entrySet())
```

Comment:
If system properties are modified I'd guess you could still get a concurrent modification error from the hash set constructor. Perhaps this class should create normalized copies of system properties and env in static initializer.

Reply:
You are right! I modified the commit a bit in order to access the immutable set of property names that `Properties` provides. Let me know what you think! :)